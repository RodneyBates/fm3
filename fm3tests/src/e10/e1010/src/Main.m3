MODULE Main

(* Illegal recursive declarations. *)

; TYPE R = REF R (* This is legal. *)

; TYPE A = ARRAY OF A 

; TYPE D = ARRAY OF E

; TYPE B = ARRAY OF C 

; TYPE Rec = RECORD
      F1 : Rec 
    ; F2 : Rec
    END 

; TYPE C = ARRAY OF D  

; TYPE E = ARRAY OF B  

; TYPE S = REF S (* This is legal. *)

; BEGIN
  END Main
. 