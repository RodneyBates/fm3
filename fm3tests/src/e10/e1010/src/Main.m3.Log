[31mERROR: [0mMain.m3 Unable to locate exported source file Main.i3
       Search directories for source files are:
       "."
[31mERROR: [0mMain.m3[7,7] Illegal recursive declaration: "A"
[31mERROR: [0mMain.m3[9,7] Illegal recursive declarations: "D", also involving
       "E" [20,7]
       "B" [11,7]
       "C" [18,7]
[31mERROR: [0mMain.m3[13,7] Illegal recursive declaration: "Rec"
