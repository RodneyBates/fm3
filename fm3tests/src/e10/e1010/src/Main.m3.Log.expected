[32mINFO:  [0mMain.m3 Pass 1 output file Main.m3.FM3Pass1 has 675 bytes.
[31mERROR: [0mMain.m3[7,7] Illegal recursive declaration: "A"
[31mERROR: [0mMain.m3[9,7] Illegal recursive declarations: "D", also involving
       "E" [20,7]
       "B" [11,7]
       "C" [18,7]
[31mERROR: [0mMain.m3[13,7] Illegal recursive declaration: "Rec"
[32mINFO:  [0mMain.m3 Patch stack Main.m3.FM3Patch peak size = 71 bytes.
[32mINFO:  [0mMain.m3 Pass 2 output file Main.m3.FM3Pass2 has 387 bytes.
